test/e2e/basic_workflow_test.go

```diff
@@ -65,121 +67,76 @@ func TestBasicWorkflow(t *testing.T) {
 
 func testServiceListEmpty(t *testing.T, k kn) {
 	out, err := k.RunWithOpts([]string{"service", "list"}, runOpts{NoNamespace: false})
-	if err != nil {
-		t.Fatalf(fmt.Sprintf("Error executing 'kn service list' command. Error: %s", err.Error()))
-	}
+	assert.NilError(t, err)
 
-	if !strings.Contains(out, "No resources found.") {
-		t.Fatalf("Expected output 'No resources found.' Instead found:\n%s\n", out)
-	}
+	assert.Check(t, util.ContainsAll(out, "No resources found."))
 }
 
 func testServiceCreate(t *testing.T, k kn, serviceName string) {
 	out, err := k.RunWithOpts([]string{"service", "create",
 		fmt.Sprintf("%s", serviceName),
 		"--image", KnDefaultTestImage}, runOpts{NoNamespace: false})
-	if err != nil {
-		t.Fatalf(fmt.Sprintf("Error executing 'kn service create' command. Error: %s", err.Error()))
-	}
+	assert.NilError(t, err)
 
-	if !strings.Contains(out, fmt.Sprintf("Service '%s' successfully created in namespace '%s'.", serviceName, k.namespace)) {
-		t.Fatalf(fmt.Sprintf("Expected to find: Service '%s' successfully created in namespace '%s'. Instead found:\n%s\n", serviceName, k.namespace, out))
-	}
+	assert.Check(t, util.ContainsAll(out, "Service", serviceName, "successfully created in namespace", k.namespace, "OK"))
 }
 
 func testServiceList(t *testing.T, k kn, serviceName string) {
 	out, err := k.RunWithOpts([]string{"service", "list", serviceName}, runOpts{NoNamespace: false})
-	if err != nil {
-		t.Fatalf(fmt.Sprintf("Error executing 'kn service list %s' command. Error: %s", serviceName, err.Error()))
-	}
+	assert.NilError(t, err)
 
 	expectedOutput := fmt.Sprintf("%s", serviceName)
-	if !strings.Contains(out, expectedOutput) {
-		t.Fatalf("Expected output incorrect, expecting to include:\n%s\n Instead found:\n%s\n", expectedOutput, out)
-	}
+	assert.Check(t, util.ContainsAll(out, expectedOutput))
 }
 
 func testRevisionListForService(t *testing.T, k kn, serviceName string) {
 	out, err := k.RunWithOpts([]string{"revision", "list", "-s", serviceName}, runOpts{NoNamespace: false})
-	if err != nil {
-		t.Fatalf(fmt.Sprintf("Error executing 'kn revision list -s %s' command. Error: %s", serviceName, err.Error()))
-	}
-	outputLines := strings.Split(out, "\n")
-	for _, line := range outputLines[1:] {
-		if len(line) > 1 && !strings.HasPrefix(line, serviceName) {
-			t.Fatalf(fmt.Sprintf("Expected output incorrect, expecting line to start with service name: %s\nFound: %s", serviceName, line))
-		}
-	}
+	assert.NilError(t, err)
+
+	assert.Check(t, util.ContainsAll(out, serviceName, "True"))
```

Comment:
Not sure if this is equivalent to the previous behaviour.
Tbh, I'm not sure whether the test before really worked as intended, as the revsion list now starts with the revsion names, not the service name (this is in the second column). 
Also you should test that *every* line (starting from line 2) contains the service name.

Not sure what the "True" comes from.

Reply:
I think the test before is not working as expected but passed because service name is included in the part of revision name. Also, I intentionally changed the logic to check if both service name and status of the revision ("True") are contained in the full output text. But, I will follow your suggestion to avoid missing the verification in case of multiple revisions associated with a single service.

Reply:
Good catch.

Maybe you could check for space before/and after the serviceName ? Like in `" " + serviceName + " "` ? That way you can be ensure that the serviceName is included.

But I would still iterate over all line and ensure that the condition is true for every line.